scripts/validate_map.py

```diff
@@ -62,27 +100,103 @@ def check_code_name_absent(df: pd.DataFrame, syn: Synapse, config: dict) -> list
-        code_remove = []
-        for code in code_map:
-            if bool(re.match(r"^.+[*]$", str(code).strip())):
-                code_remove.append(code)
-            elif pd.isna(code):
-                code_remove.append(code)
+        code_remove = get_codes_to_remove(code_map)
         for code in code_remove:
             code_map.remove(code)
 
         absent.extend(list(set(code_map) - set(code_data)))
     return absent
 
 
-def format_result(codes: list, config: dict, check_no: int) -> pd.DataFrame:
+def check_dataset_names(
+    df: pd.DataFrame, syn: Synapse, config: Dict, cohort: str, release: str
+) -> List:
+    """Check for any dataset name that is not associated with a dataset.
+
+    Args:
+        df (pd.DataFrame): dataframe representing map
+        syn (Synapse): Synapse object
+        config (dict): configuration parameters
+        cohort (str): cohort label to check
+        release (str): release label to check
+
+    Returns:
+        list: dataset names in map but not on synapse
+    """
+
+    query = f'SELECT DISTINCT dataset FROM {config["synapse"]["dataset"]["id"]} WHERE dataset IS NOT NULL'
+    table_ds = syn.tableQuery(query).asDataFrame()
+    map_ds = df["dataset"].unique()
+    res = set([x for x in map_ds if pd.isnull(x) == False]) - set(table_ds["dataset"])
+    return list(res)
+
+
+def check_release_status_ambiguous(
+    df: pd.DataFrame, syn: Synapse, config: Dict, cohort: str, release: str
+) -> List:
+    """Check for any codes with release status that is not y or n.
+
+    Args:
+        df (pd.DataFrame): dataframe representing map
+        syn (Synapse): Synapse object
+        config (dict): configuration parameters
+        cohort (str): cohort label to check
+        release (str): release label to check
+
+    Returns:
+        list: codes with ambiguous release status
+    """
+    status = df[cohort].str.lower()
+    codes = df.loc[[item not in ["y", "n"] for item in status]]["code"]
+    return codes
+
+
+def check_release_status_map_yes_sor_not(
+    df: pd.DataFrame, syn: Synapse, config: Dict, cohort: str, release: str
+) -> List:
+    """Check for codes where release status in mapping file is yes
+    but relase status in scope of release is not yes.
+
+    Args:
+        df (pd.DataFrame): dataframe representing map
+        syn (Synapse): Synapse object
+        config (dict): configuration parameters
+        cohort (str): cohort label to check
+        release (str): release label to check
+
+    Returns:
+        list: codes with lenient release status
+    """
+    map_status = df[cohort].str.lower()
+    map_type = df["data_type"].str.lower()
+    map_codes = df.loc[
+        ((map_status == "y") & ((map_type == "derived") | (map_type == "curated")))
+    ]["code"]
+
+    column_name = config["column_name"]["sor_cbio"][cohort][release]
+    file_sor = syn.get(config["synapse"]["sor"]["id"])["path"]
+    sor = pd.read_excel(file_sor, engine="openpyxl", sheet_name=1)
```

Comment:
Why openpyxl?

Reply:
Default engine for pandas.read_excel is xlrd and doesn't support reading xlsx, which is what file_sor is.  

https://stackoverflow.com/questions/65254535/xlrd-biffh-xlrderror-excel-xlsx-file-not-supported